### PR TITLE
Fixing Support for EMR

### PR DESCRIPTION
**Pull request by ahmadnazeri**
# Description
Goal is to fix the support for EMR and Databricks. 

### Select type(s) of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have fixed any merge conflicts

### CODE REVIEW

File: client/src/featureform/register.py [hunk 1]
```diff
@@ -20,14 +20,18 @@
...
     SQLTransformation, DFTransformation, Entity, Feature, Label, ResourceColumnMapping, TrainingSet, ProviderReference, \
-    EntityReference, SourceReference, ExecutorCredentials, ResourceRedefinedError, ResourceStatus, Transformation
+    EntityReference, SourceReference, ExecutorCredentials, ResourceRedefinedError, ResourceStatus, Transformation, AWSCredentials
 
 from .proto import metadata_pb2_grpc as ff_grpc
 
+
 NameVariant = Tuple[str, str]
 
+s3_config = S3StoreConfig("", "", AWSCredentials("id", "secret"))
```
Comment (sdreyer) [inline client/src/featureform/register.py:32]: Is there a way to do this without having to create the actual object?

File: client/src/featureform/resources.py [hunk 2]
```diff
@@ -104,6 +104,29 @@ def serialize(self) -> bytes:
...
 
 
+@typechecked
+@dataclass
+class AWSCredentials:
+    def __init__(self,
+                 aws_access_key_id: str = "",
+                 aws_secret_access_key: str = "",):
+        empty_strings = aws_access_key_id == "" or aws_secret_access_key == ""
+        assert not empty_strings, Exception(
```
Comment (sdreyer) [inline client/src/featureform/resources.py:114]: Are asserts typically used in non-testing situations?

File: client/src/featureform/register.py (hunk 1 above)
Comment (ahmadnazeri) [reply client/src/featureform/register.py:32]: We can hardcode the types but we will have to modify it if we modify the type.

File: client/src/featureform/resources.py (hunk 2 above)
Comment (ahmadnazeri) [reply client/src/featureform/resources.py]: Modified it to exceptions.

File: client/src/featureform/register.py (hunk 1 above)
Comment (sdreyer) [reply client/src/featureform/register.py:32]: Does it make sense to make all the types an enum?

File: client/src/featureform/register.py [hunk 3]
```diff
@@ -3681,4 +3743,5 @@ def list_providers(self, local=False):
 get_spark_aws = global_registrar.get_spark
```
Comment (sdreyer) [inline client/src/featureform/register.py:3743]: This should just be "get_spark"

File: client/tests/test_spark_provider.py [hunk 4]
```diff
@@ -6,20 +6,30 @@
...
+    executor = request.getfixturevalue(executor_fixture)
+    filestore = request.getfixturevalue(filestore_fixture)
 
-    databricks = DatabricksCredentials(username="a", password="b", cluster_id="c_id")
-    azure_blob = AzureFileStoreConfig(account_name="", account_key="", container_name="", root_path="")   
+    provider_name = "test_offline_spark_provider"
+    r = Registrar() 
     
-    config = SparkConfig(executor_type=databricks.type(), executor_config=databricks.config(), store_type=azure_blob.store_type(), store_config=azure_blob.config())
+    config = SparkConfig(executor_type=executor.type(), executor_config=executor.config(), store_type=filestore.store_type(), store_config=filestore.config())
```
Comment (sdreyer) [inline client/tests/test_spark_provider.py:25]: Is there a check that register_spark() will typecheck/accept these types?

Comment (ahmadnazeri) [reply client/tests/test_spark_provider.py:25]: The `register_spark` method requires a `executor: ExecutorCredentials` and `filestore: FileStoreProvider`. That's the check on the type.

Comment (sdreyer) [reply client/tests/test_spark_provider.py:25]: Ahh the register functions return those. All good

**Pull request merged**